localgraphclustering/src/lib/set_scores.cpp

```diff
@@ -0,0 +1,100 @@
+/* Compute cut size, number of edges given a set and a graph */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unordered_map>
+#include <queue>
+#include <iostream>
+#include <algorithm>
+#include <stdint.h>
+
+#include "include/setscores_c_interface.h"
+#include "include/routines.hpp"
+
+using namespace std;
+
+void set_scores32(
+		uint32_t n, uint32_t* ai, uint32_t* aj, uint32_t offset,
+		uint32_t* R, uint32_t nR, uint32_t* voltrue, uint32_t* cut)
+{
+	graph<uint32_t,uint32_t> g(ai[n],n,ai,aj,NULL,offset,NULL);
+    unordered_map<uint32_t, uint32_t> R_map;
+    for (size_t i = 0; i < nR; i ++) {
```

Comment:
This for loop is duplicated in each of the calls. Can you turn it into a function?